masonry/src/render_root.rs

```diff
@@ -159,21 +167,23 @@ impl RenderRoot {
             rebuild_access_tree: true,
         };
 
-        // We run a set of passes to initialize the widget tree
-        run_update_new_widgets_pass(&mut root);
-        // TODO - Remove this line
-        let mut dummy_state = WidgetState::synthetic(root.root.id(), root.get_kurbo_size());
-        root.post_event_processing(&mut dummy_state);
-
-        // We run a layout pass right away to have a SetSize signal ready
-        if size_policy == WindowSizePolicy::Content {
-            root.root_layout();
+        if let Some(test_font_data) = test_font {
+            let families = root.register_fonts(test_font_data);
+            // Make sure that all of these fonts are in the fallback chain for the Latin script.
+            // <https://en.wikipedia.org/wiki/Script_(Unicode)#Latn>
+            root.state
+                .font_context
+                .collection
+                .append_fallbacks(*b"Latn", families.iter().map(|(family, _)| *family));
         }
 
+        // We run a set of passes to initialize the widget tree
+        root.run_rewrite_passes();
```

Comment:
It's a bit weird to be doing a layout pass here, since the size of the window doesn't meaningfully exist yet.

I suspect some of this lifecycle/ordering things might need to be thought through very carefully.
See also #590. OTOH, this isn't urgent to fix, so this version is fine.

Reply:
I think it acts as a decent smoke test for layout bugs that lead to crashes when size constraints are zero. And we do want to detect those bugs early.